client/src/containers/edit-user/user-form.tsx

```diff
@@ -0,0 +1,96 @@
+import { useMemo, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { omit, pick } from 'lodash-es';
+
+import { Button } from 'components/button';
+import Input from 'components/forms/input';
+import Select from 'components/forms/select';
+import { RoleName } from 'hooks/permissions/enums';
+
+import type { User } from 'types';
+import type { Option } from 'components/forms/select';
+
+type UserFormData = Pick<User, 'fname' | 'lname' | 'displayName' | 'email'> & {
+  roles: RoleName[];
+};
+
+type UserFormProps = {
+  user: User;
+  children: React.ReactNode;
+};
+
+const UserForm = ({ user, children }: UserFormProps) => {
+  const { register, control, setValue } = useForm<UserFormData>({
+    defaultValues: {
+      ...pick(user, 'fname', 'lname', 'displayName', 'email'),
+      roles: user.roles.map((role) => role.name),
+    },
+  });
+
+  const roleOptions = useMemo(
+    () => Object.values(RoleName).map((role) => ({ label: role, value: role })),
+    [],
+  );
+
+  const [roleValue, setRoleValue] = useState<Option<string>[]>(
+    user.roles.map((role) => ({ label: role.name, value: role.name })),
+  );
```

Comment:
```suggestion
  const [roleValue, setRoleValue] = useState<Option<string>[]>(roleOptions);
```
Couldn't be like this?

Reply:
It can’t because the  initial state is related to the user data coming from the API, and the roleOptions is a hardcoded value